libraries/AP_HAL_ChibiOS/GPIO.cpp

```diff
@@ -486,8 +486,11 @@ bool GPIO::wait_pin(uint8_t pin, INTERRUPT_TRIGGER_TYPE mode, uint32_t timeout_u
         osalSysUnlock();
         return false;
     }
-        
-    msg_t msg = osalThreadSuspendTimeoutS(&g->thd_wait, TIME_US2I(timeout_us));
+
+    // don't allow for very long timeouts, or below the delta
+    timeout_us = constrain_uint32(TIME_US2I(timeout_us), CH_CFG_ST_TIMEDELTA, TIME_US2I(30000U));
```

Comment:
Makes me itch a little bit - but us2i is 1-1 ATM, so *shrug*.

Reply:
It makes me itch as well, feels like a bug waiting to happen is some distant future - why not fix it now while its easy?